doxygen build in sandbox fails

Hey, 

I can't built `devel::doxygen` in sandbox with the following error message:

```
CMake Error at /bob/57ddf4ba5fedc6cc2911e0f7ffca701259a88072/workspace/usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Iconv (missing: ICONV_TEST ICONV_INCLUDE_DIR)
Call Stack (most recent call first):
  /bob/57ddf4ba5fedc6cc2911e0f7ffca701259a88072/workspace/usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindIconv.cmake:87 (find_package_handle_standard_args)
  CMakeLists.txt:120 (find_package)
```

this is fixed in with this for me: https://github.com/BobBuildTool/basement/pull/78/files 
https://github.com/BobBuildTool/basement/blob/d440a34f65b4695acfed9d1673923af2091b54fd/classes/cmake.yaml#L14-L18

is this fine for an integration into the master?

--- a/classes/cmake.yaml
+++ b/classes/cmake.yaml
@@ -12,6 +12,14 @@ buildSetup: |
         CMAKE_FIND_ROOT_PATH+="${CMAKE_FIND_ROOT_PATH:+;}$i"
     done
 
+    # It is necessary to add the sysroot path of gcc/g++ if available too
+    # because our host-compat-toolchain has a non-standard sysroot
+    # even though it's a native compiler
+    if [[ "$CC" == "gcc" ]] ; then
+        SYSROOT="$($CC -print-sysroot)"
+    fi
+    [ -z "${SYSROOT:+true}" ] || CMAKE_FIND_ROOT_PATH+="${CMAKE_FIND_ROOT_PATH:+;}${SYSROOT}"
+
     # This looks odd but it prevents MSYS from converting /usr to C:\msys64\usr
     # (or wherever MSYS was installed). See
     # http://mingw.org/wiki/Posix_path_conversion

